### PR TITLE
Add moveRows to Query module

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
Related PR added a query-moveRows.api to the QueryController so that we could consolidate the various move entities actions to a single action. This PR adds moveRows() to the Query module in @labkey/api.

#### Related Pull Requests
- https://github.com/LabKey/labkey-ui-components/pull/1369
- https://github.com/LabKey/labkey-ui-components/pull/1372

#### Changes
- Add `moveRows` to `Query` module, along with related types: `MoveRowsOptions` and `MoveRowsResponse`

### CODE REVIEW

File: src/labkey/query/Rows.ts [hunk 1]
```diff
@@ -232,6 +232,17 @@ export interface Command {
     schemaName: string;
 }
 
+export interface  MoveRowsResponse extends ModifyRowsResults {
+    /** The container path in which the rows were moved. */
+    containerPath: string;
```
Comment (labkey-nicka) [inline src/labkey/query/Rows.ts]: Should `containerPath` really be marked as required?

File: src/labkey/query/Rows.ts [hunk 2]
```diff
@@ -413,3 +424,48 @@ export function truncateTable(options: QueryRequestOptions): XMLHttpRequest {
...
+     * The data region selectionKey to use to get the rowIds to be moved if not rows param is provided.
+     */
+    dataRegionSelectionKey?: string;
+    /**
+     * The target container in which the rows should be moved.
+     */
+    targetContainerPath: string;
+    /**
+     * Used in conjunction with dataRegionSelectionKey to specify whether to use the current selection or the snapshot selection.
+     * */
```
Comment (labkey-nicka) [inline src/labkey/query/Rows.ts:440]: nit: end of comment wrapping

**Review by labkey-nicka — APPROVED** [review]

File: src/labkey/query/Rows.ts (hunk 1 above)
Comment (cnathe) [reply src/labkey/query/Rows.ts]: Changed to optional.

File: src/labkey/query/Rows.ts (hunk 2 above)
Comment (cnathe) [reply src/labkey/query/Rows.ts:440]: done

**Pull request merged**